File: simple-stack-example-basic/src/main/java/com/zhuinden/simplestackdemoexample/common/SecondKey.java
package com.zhuinden.simplestackdemoexample.common;

import android.support.annotation.NonNull;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestack.navigator.ViewChangeHandler;
import com.zhuinden.simplestack.navigator.changehandlers.NoOpViewChangeHandler;
import com.zhuinden.simplestackdemoexample.R;
/**
 * Created by Owner on 2017. 01. 12..
 */
@AutoValue // parcelable, hashcode, equals, toString
public abstract class SecondKey
        implements Key {
    public static SecondKey create() {
        return new AutoValue_SecondKey();
    }

    @Override
    public int layout() {
        return R.layout.path_second;
    }

    @NonNull
    @Override
    public ViewChangeHandler viewChangeHandler() {
        return new NoOpViewChangeHandler();
    }
}

File: simple-stack-example-basic-kotlin/src/main/kotlin/com/zhuinden/simplestackdemoexample/common/SecondKey.kt
package com.zhuinden.simplestackdemoexample.common

import com.zhuinden.simplestack.navigator.ViewChangeHandler
import com.zhuinden.simplestack.navigator.changehandlers.NoOpViewChangeHandler
import com.zhuinden.simplestackdemoexample.R
import paperparcel.PaperParcel
import paperparcel.PaperParcelable

/**
 * Created by Owner on 2017. 01. 12..
 */
@PaperParcel
data class SecondKey protected constructor(protected val tag: String = SecondKey::javaClass.name) //
    : PaperParcelable, Key {
    override fun layout(): Int {
        return R.layout.path_second
    }

    override fun viewChangeHandler(): ViewChangeHandler {
        return NoOpViewChangeHandler()
    }

    companion object {
        @JvmField val CREATOR = PaperParcelSecondKey.CREATOR

        fun create(): SecondKey {
            return SecondKey()
        }
    }
}
